xbmc/video/VideoInfoScanner.cpp

```diff
@@ -818,6 +886,72 @@ namespace VIDEO
     return INFO_NOT_FOUND;
   }
 
+  bool CVideoInfoScanner::UpdateSetInfo(CFileItem* pItem)
+  {
+    // If there is a set, see if the details need to be updated from the
+    // Movie Set Information Folder
+    bool setUpdated = false;
+    if (!pItem->GetVideoInfoTag()->m_set.title.empty())
+    {
+      const std::string movieSetInfoPath =
+          VIDEO::CVideoInfoScanner::GetMovieSetInfoFolder(pItem->GetVideoInfoTag()->m_set.title);
+      if (!movieSetInfoPath.empty())
+      {
+        // look for Set.NFO
+        std::unique_ptr<ISetInfoTagLoader> setLoader;
+        AddonPtr addon{};
+        CServiceBroker::GetAddonMgr().GetAddon("metadata.local", addon, ADDON::OnlyEnabled::CHOICE_YES);
+        ScraperPtr scraper = std::dynamic_pointer_cast<CScraper>(addon);
+        setLoader.reset(CSetInfoTagLoaderFactory::CreateLoader(*pItem, scraper));
+        if (setLoader)
+        {
+          CSetInfoTag s{};
+          CInfoScanner::INFO_TYPE result{CInfoScanner::NO_NFO};
+          result = setLoader->Load(s, false);
+          if (!s.IsEmpty())
+          {
+            if (!s.GetTitle().empty())
+              pItem->GetVideoInfoTag()->m_set.title = s.GetTitle();
+            if (!s.GetOverview().empty())
+              pItem->GetVideoInfoTag()->m_set.overview = s.GetOverview();
```

Comment:
You are calling `pItem->GetVideoInfoTag()` several time sin this method. If you look at its implementation you will find that it is not inline/trival. I would introduce a variable to hold the infotag's value.